flink-architecture-tests/flink-architecture-tests-production/archunit-violations/b8900323-6aab-4e7e-9b17-f53b3c3dca46

```diff
@@ -17,11 +17,10 @@ Method <org.apache.flink.connector.hbase.sink.HBaseSinkFunction.prepareRuntimeCo
 Method <org.apache.flink.connector.hbase.sink.HBaseSinkFunction.prepareRuntimeConfiguration()> has return type <org.apache.hadoop.conf.Configuration> in (HBaseSinkFunction.java:0)
 Method <org.apache.flink.connector.hbase.source.HBaseRowDataLookupFunction.prepareRuntimeConfiguration()> calls method <org.apache.hadoop.conf.Configuration.get(java.lang.String)> in (HBaseRowDataLookupFunction.java:147)
 Method <org.apache.flink.connector.hbase.source.HBaseRowDataLookupFunction.prepareRuntimeConfiguration()> has return type <org.apache.hadoop.conf.Configuration> in (HBaseRowDataLookupFunction.java:0)
-Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:113)
-Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:114)
+Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:101)
+Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:102)
+Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:108)
 Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> has return type <org.apache.hadoop.conf.Configuration> in (HBaseConnectorOptionsUtil.java:0)
-Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.lambda$getHBaseConfiguration$0(org.apache.hadoop.conf.Configuration, java.lang.Object, java.lang.Object)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:120)
-Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.lambda$getHBaseConfiguration$0(org.apache.hadoop.conf.Configuration, java.lang.Object, java.lang.Object)> has parameter of type <org.apache.hadoop.conf.Configuration> in (HBaseConnectorOptionsUtil.java:0)
```

Comment:
We seem to have lost this violation.

Reply:
this line is disappearing after applying this commit done under https://github.com/TNG/ArchUnit/pull/847
It looks like the merge 2 lambda refs (for this case) into one
within the code `org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil#getHBaseConfiguration`
there are 3 `set` methods
```java
    public static Configuration getHBaseConfiguration(ReadableConfig tableOptions) {
        // create default configuration from current runtime env (`hbase-site.xml` in classpath)
        // first,
        Configuration hbaseClientConf = HBaseConfigurationUtil.getHBaseConfiguration();
        hbaseClientConf.set(HConstants.ZOOKEEPER_QUORUM, tableOptions.get(ZOOKEEPER_QUORUM));
        hbaseClientConf.set(
                HConstants.ZOOKEEPER_ZNODE_PARENT, tableOptions.get(ZOOKEEPER_ZNODE_PARENT));
        // add HBase properties
        final Properties properties =
                getHBaseClientProperties(
                        ((org.apache.flink.configuration.Configuration) tableOptions).toMap());
        properties.forEach((k, v) -> hbaseClientConf.set(k.toString(), v.toString()));
        return hbaseClientConf;
    }
```
and now there are only four lines for that
```
Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:101)
Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:102)
Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> calls method <org.apache.hadoop.conf.Configuration.set(java.lang.String, java.lang.String)> in (HBaseConnectorOptionsUtil.java:108)
Method <org.apache.flink.connector.hbase.table.HBaseConnectorOptionsUtil.getHBaseConfiguration(org.apache.flink.configuration.ReadableConfig)> has return type <org.apache.hadoop.conf.Configuration> in (HBaseConnectorOptionsUtil.java:0)
```

still not clear why it was one more before that... I checked other classes with lambda e.g. `org.apache.flink.connector.jdbc.internal.converter.OracleRowConverter` and there is no such behavior...
Will have a look a bit more may be will find something

Reply:
AFAICT this hasn't got anything to do with "merging" errors.

The lambda issue was for having a Hadoop configuration _parameter_, a different error altogether, and that is now gone.

I would assume that the parameter type check is just no longer applied to lambdas, presumably because when you define such a lambda you either
a) access said type somewhere else (to pass as a parameter)
b) pass the lambda as an argument, but then you get another violation anyway.

(nevermind that this "architecture" test is pretty weird since it complains about _every single usage of external libraries_).

Reply:
IOW that the entry is gone shouldn't be a problem afaict.